homeassistant/components/ping/binary_sensor.py

```diff
@@ -261,9 +248,13 @@ async def async_ping(self):
                 match = PING_MATCHER_BUSYBOX.search(
                     str(out_data).rsplit("\n", maxsplit=1)[-1]
                 )
+                if match is None:
+                    return None
```

Comment:
Is this a bugfix?

Reply:
No, only to make mypy happy

Reply:
You mean because match is reused as local variable? May be you should type it?

Reply:
```suggestion
                assert match is not None
```